autoload/ale/sign.vim

```diff
@@ -67,10 +67,14 @@ endfunction
 " Given a loclist, combine the loclist into a list of signs such that only
 " one sign appears per line. Error lines will take precedence.
 " The loclist will have been previously sorted.
-function! ale#sign#CombineSigns(loclist) abort
+function! ale#sign#CombineSigns(buffer, loclist) abort
     let l:signlist = []
 
     for l:obj in a:loclist
+        if has_key(l:obj, 'bufnr') && l:obj.bufnr >= 0 && l:obj.bufnr != a:buffer
```

Comment:
That's not the way quickfix/location list work. Go will produce errors for any file in the package. Using loclist is a great way to quickly jump to the line in another file that is causing the package the first file is in not to compile.

The errors show up properly in loclist, this is great. Discarding these errors in the lint handler is taking away useful information from the user.

The problem is that without the change I put in this file, signs from errors/warnings in other buffers get put in the linted file's buffer and shows that there is an error that doesn't exist.